### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by vrx29**
This PR contains wishlist logic for the ecom app. Private routing is also enable so you cannot access cart and wishlist without loggin in into the app.
You can signup or use my credentials

Credentials -

id : [vineet@gmail.com](mailto:vineet@gmail.com)
pass: vineet123

As there are multiple files you can review these files only.
:frog: [wishlist context](https://github.com/vrx29/gimesto-store-ecom/pull/5/files#diff-eca5fc35b179a6370adf4591e52549cfce0e7ad7c85d367b7b59c0a2b640ed1a)
:frog: [Wishlist Page](https://github.com/vrx29/gimesto-store-ecom/pull/5/files#diff-e8dc074032247d88aad3ef5decba37f783a97759ffa2fc6205b565d59b112395)
:frog: [Wishlist Card](https://github.com/vrx29/gimesto-store-ecom/pull/5/files#diff-4a44a1b6246a3b4c24e71b8cc4f1ffab9e7ede57c47d0402f5f412f87eed8d52)
:frog: [Product Card](https://github.com/vrx29/gimesto-store-ecom/pull/5/files#diff-872c29d4c429674e938143f04b07b54098345d7a7ffde7bfa46b406304d74d3d)

### CODE REVIEW

File: src/context/wishListContext.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import { createContext, useContext, useReducer } from "react";
+import { useFetchApi } from "../hooks/useFetchApi";
+import { useAuth } from "./";
+import { initialSharedState } from "../reducers/constants";
+import { sharedReducer } from "../reducers/reducerFunctions/sharedReducer";
+import jwt from "jsonwebtoken";
```
Comment (hsnice16) [inline src/context/wishListContext.js]: I think you have not used this `jwt` anywhere so you can remove this import statement.

File: src/context/wishListContext.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  );
+  const apiData = {
+    api: "/api/user/wishlist",
+    property: "wishList",
+  };
+  const { userAuthState } = useAuth();
+  const { authToken } = userAuthState;
+
+  const { state } = useFetchApi(apiData, {
+    headers: {
```
Comment (hsnice16) [inline src/context/wishListContext.js:23]: This `headers` configuration is getting repeated three times, so you can create a variable and then can use that. Like

```javascript
const config = {
         headers: {
            authorization: authToken,
          },
}
```

And then, can use like
```javascript
await axiost.get(api, config)
```

File: src/context/wishListContext.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      const res = await axios.delete(`/api/user/wishlist/${productId}`, {
+        headers: {
+          authorization: authToken,
+        },
+      });
+      wishlistDispatch({ type: "SET_DATA", payload: res.data.wishlist });
+    } catch (error) {
+      wishlistDispatch({
+        type: "SET_ERROR",
+        payload: "Error occured while adding item to wishlist",
```
Comment (hsnice16) [inline src/context/wishListContext.js]: Here, I think the `payload` should be `while deleting`

**Review by hsnice16 — APPROVED** [review]
Nice work, file structuring is good. The code is clean too. Just have added a few suggestions.

File: src/context/wishListContext.js (hunk 1 above)
Comment (vrx29) [reply src/context/wishListContext.js]: Yes I forgot to remove this will remove it now.

File: src/context/wishListContext.js (hunk 2 above)
Comment (vrx29) [reply src/context/wishListContext.js:23]: will check this

File: src/context/wishListContext.js (hunk 3 above)
Comment (vrx29) [reply src/context/wishListContext.js]: Yes I did copy paste wishlist code. So forgot to change this

File: src/context/wishListContext.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import { sharedReducer } from "../reducers/reducerFunctions/sharedReducer";
+import axios from "axios";
+
+const WishListContext = createContext();
+
+const WishListProvider = ({ children }) => {
+  const [wishlistState, wishlistDispatch] = useReducer(
+    sharedReducer,
+    initialSharedState
+  );
```
Comment (anik31) [inline src/context/wishListContext.js:14]: appreciation - good that you have created a different context for wishlist

File: src/context/wishListContext.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  const addToWishlist = async (product) => {
+    try {
+      const res = await axios.post(
+        "/api/user/wishlist",
+        {
+          product,
+        },
+        config
+      );
+      wishlistDispatch({ type: "SET_DATA", payload: res.data.wishlist });
```
Comment (anik31) [inline src/context/wishListContext.js:38]: suggestion - you can add check for success status code & then dispatch

File: src/pages/WishList/WishList.js [hunk 6]
```diff
@@ -1,15 +1,35 @@
...
+              item={item}
+              removeWishlist={deleteFromWishlist}
+            />
+          ))
+        ) : (
+          <div className="cart-empty">
+            <p className="wishlist-msg">
+              You wishlist is empty. Please add some items!
+            </p>
+            <img src={wishlistEmpty} alt="Empty cart" />
```
Comment (anik31) [inline src/pages/WishList/WishList.js:30]: appreciation - nice, you have added message for empty wishlist

**Review by anik31 — APPROVED** [review]
Nice work @vrx29 💯  Code is clean & easily readable. UI is also working fine.

File: src/context/wishListContext.js (hunk 5 above)
Comment (vrx29) [reply src/context/wishListContext.js:38]: sure will check this

**Review by BalajiHarinath — APPROVED** [review]
Your code is very clean. Awesome work.

**Pull request merged**